acme/environment_loops/open_spiel_environment_loop.py

```diff
@@ -0,0 +1,247 @@
+# python3
+# Copyright 2018 DeepMind Technologies Limited. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""An OpenSpiel multi-agent/environment training loop."""
+
+import operator
+import time
+from typing import List, Optional
+
+from acme import core
+# Internal imports.
+from acme.tf import utils as tf2_utils
+from acme.utils import counting
+from acme.utils import loggers
+import dm_env
+from dm_env import specs
+import numpy as np
+import pyspiel
+import tensorflow as tf
+import tree
+
+
+class OpenSpielEnvironmentLoop(core.Worker):
+  """An OpenSpiel RL environment loop.
+
+  This takes `Environment` and list of `Actor` instances and coordinates their
+  interaction. Agents are updated if `should_update=True`. This can be used as:
+
+    loop = EnvironmentLoop(environment, actors)
+    loop.run(num_episodes)
+
+  A `Counter` instance can optionally be given in order to maintain counts
+  between different Acme components. If not given a local Counter will be
+  created to maintain counts between calls to the `run` method.
+
+  A `Logger` instance can also be passed in order to control the output of the
+  loop. If not given a platform-specific default logger will be used as defined
+  by utils.loggers.make_default_logger. A string `label` can be passed to easily
+  change the label associated with the default logger; this is ignored if a
+  `Logger` instance is given.
+  """
+
+  def __init__(
+      self,
+      environment: dm_env.Environment,
+      actors: List[core.Actor],
```

Comment:
This could be Sequence[core.Actor] right?

Reply:
Fixed